javascript.md

```diff
@@ -2,8 +2,87 @@
 layout: page
 title: Javascript
 ---
-TODO
 
-# ChangeLog
+## Start here
 
-Initial Placeholder
+One of the best comunity driven resources for coding standards is [Steve Kwan's best practices][steve-kwan-best-practices]. The recommended reading is excellent and it's well worth following the links therein.
+
+## Recommended Reading
+
+This one is already mentioned in [Steve Kwan's best practices][steve-kwan-best-practices] but is worth mentioning twice since:
+
+[Javascript - The Good Parts][javascript-the-good-parts] (Douglas Crockford's)
+
+[Learning Javascript Design Patterns][learning-javascript-design-patterns] (Addy Osmani)
+
+## General Tips
+
+- Get peer reviews of your code: Make 'em short; do them early; small reviews get much faster feedback.
+- New files need to be added to the build profile.
+- When native javascript APIs exist, use these instead of those provided by a framework. i.e. Native `array.forEach` vs `dojo.base.array.forEach`.
+- Similarly, whenever possible, use native HTML5 widgets over framework widgets such as dojo's `dijit`. Only use other widgets when absolutely necessary.
+- Use [JSDoc3][jsdoc3] to document your code.
+- Write *TESTS*! Especially for your models
+- Define your CSS at component level instead of page (and put each into it's own component file)
+
+## Syntax
+
+### Use AMD syntax
+
+for all new work. When opportunity allows, refactor legacy files to use AMD syntax. Typically this is when you need to make changes to them.
```

Comment:
Personal style: I'd prefer the heading to be repeated on the first line. Like:

```
### Use AMD Syntax

Use AMD Syntax for all new work. When opportunity...
```

Otherwise the original reads to me as though you've left out the start of the sentence (this is indicated by the start being lower case).